### PR TITLE
DS-665 - Add logic to prevent Boolean values within JSON being sent as strings

### PR DESCRIPTION
**Pull request by martin-taylor-visitscotland**

### CODE REVIEW

**Review by jlrhurta — APPROVED** [review]

File: src/components/form/Form.vue [hunk 1]
```diff
@@ -567,14 +567,22 @@ export default {
...
-            hiddenInputFields.forEach((field) => {
-                fieldData[field.name] = field.value;
-            });
+            return [...hiddenInputFields].map(this.parseBooleanStringsFromInputField);
+        },
+        parseBooleanStringsFromInputField(inputField) {
+            let value = inputField.value;
 
-            return fieldData;
+            if (value === 'true') {
```
Comment (GemmaLouise) [inline src/components/form/Form.vue:576]: This could probably written a bit neater with a ternary statement

Comment (martin-taylor-visitscotland) [reply src/components/form/Form.vue:576]: Great shout :)

Comment (martin-taylor-visitscotland) [reply src/components/form/Form.vue:576]: @GemmaLouise - I remember I did attempt this at some point but was encountering `ESLint: Do not nest ternary expressions.(no-nested-ternary`.

File: src/components/form/Form.vue [hunk 2]
```diff
@@ -567,14 +567,22 @@ export default {
...
             const hiddenInputFields = this.$el.querySelectorAll('input[type=hidden]');
-            const fieldData = {
-            };
 
-            hiddenInputFields.forEach((field) => {
-                fieldData[field.name] = field.value;
-            });
+            return [...hiddenInputFields].map(this.parseBooleanStringsFromInputField);
+        },
+        parseBooleanStringsFromInputField(inputField) {
```
Comment (GemmaLouise) [inline src/components/form/Form.vue:573]: Methods all need a comment to explain the function, we aren't using JSdocs at the moment so just a short comment to explain what its/what it returns will do

**Review by jlrhurta — APPROVED** [review]

**Review by duncansanderson — APPROVED** [review]

**Review by jcalcines — APPROVED** [review]

**Review by GemmaLouise — APPROVED** [review]

**Pull request closed without merging**